### ISSUE TITLE
ClassCastException for entity with Long primary key in Kotlin on Android

### ISSUE THREAD

**Issue by scompt** [requery/requery#693]
An example of this crash is available [here](https://github.com/scompt/requery_bug). Just build the app (`./gradlew assembleDebug`), install it (`adb install app/build/outputs/apk/debug/app-debug.apk`), run it, and you'll get the exception below.

The entity in question is defined so:

```
@Entity
data class ExampleEntity (
        @get:Key @get:Generated
        val id: Long,

        val name: String?
) : Persistable
```

and I'm trying to insert it like so:

```
        entityStore.insert(ExampleEntity(5, null), Long::class).subscribe()
```

Any ideas what could be the problem?

# Exception Stacktrace

```
10-03 12:48:37.155 E/AndroidRuntime(17918): FATAL EXCEPTION: main
10-03 12:48:37.155 E/AndroidRuntime(17918): Process: com.scompt.requery_bug, PID: 17918
10-03 12:48:37.155 E/AndroidRuntime(17918): io.reactivex.exceptions.OnErrorNotImplementedException: java.lang.Long cannot be cast to long
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at io.reactivex.internal.observers.ConsumerSingleObserver.onError(ConsumerSingleObserver.java:45)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:43)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at io.reactivex.Single.subscribe(Single.java:2703)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at io.reactivex.Single.subscribe(Single.java:2689)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at io.reactivex.Single.subscribe(Single.java:2610)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at com.scompt.requery_bug.BugApplication.onCreate(BikeBuilderApplication.kt:38)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1013)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4712)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at android.app.ActivityThread.-wrap1(ActivityThread.java)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1405)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at android.os.Handler.dispatchMessage(Handler.java:102)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at android.os.Looper.loop(Looper.java:148)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at android.app.ActivityThread.main(ActivityThread.java:5422)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at java.lang.reflect.Method.invoke(Native Method)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
10-03 12:48:37.155 E/AndroidRuntime(17918): Caused by: java.lang.ClassCastException: java.lang.Long cannot be cast to long
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at java.lang.Class.cast(Class.java:1505)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at io.requery.sql.EntityDataStore.insert(EntityDataStore.java:207)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at io.requery.sql.KotlinEntityDataStore.insert(KotlinEntityDataStore.kt:121)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at io.requery.reactivex.KotlinReactiveEntityStore$insert$3.call(KotlinReactiveEntityStore.kt:59)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:35)
10-03 12:48:37.155 E/AndroidRuntime(17918): 	... 14 more
```

**Comment by npurushe** [comment]
You should be able to work around this by using `Long::class.javaObjectType`

**Comment by scompt** [comment]
@npurushe Thanks for looking into this. I don't see where I can use `Long::class.javaObjectType`. Replacing it in the entity doesn't make sense at all and using it in the `insert` call also doesn't compile.

**Comment by k-misztal** [comment]
When is it going to be released? In the `1.4.1` (which is currently newest version) the issue is still present.
